nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/state/server/ZooKeeperStateServer.java

```diff
@@ -198,6 +216,64 @@ public static ZooKeeperStateServer create(final NiFiProperties properties) throw
             zkProperties.load(bis);
         }
 
-        return new ZooKeeperStateServer(zkProperties);
+        return new ZooKeeperStateServer(reconcileProperties(properties, zkProperties));
+    }
+
+    private static QuorumPeerConfig reconcileProperties(NiFiProperties niFiProperties, Properties zkProperties) throws IOException, ConfigException {
+        QuorumPeerConfig peerConfig = new QuorumPeerConfig();
+        peerConfig.parseProperties(zkProperties);
+
+        // If this is an insecure NiFi or if the ZooKeeper is distributed, no changes are needed:
+        if (!niFiProperties.isHTTPSConfigured() || peerConfig.isDistributed()) {
```

Comment:
I'm trying to understand the RHS of this. AFAICT this means that this will only apply to standalone with the embedded server enabled. Is this because the leader election client changes are needed before it can be enabled for a cluster?

Reply:
I think the `isDistributed` check was added incorrectly; removed.